### DISCUSSION TITLE
DuckDB writing parquet to AWS S3

### DISCUSSION THREAD

**Discussion by tylerjchrestoff** [duckdb/duckdb#10081] [General]
I have an AWS Lambda function deployed using DuckDB to read S3 files, perform some transformations, and write the output back to another S3 location as parquet. The read operation is working correctly with the Lambda's execution role but the write function is failing with a 403 forbidden error. The execution role has PutObject permissions on the bucket and there are no bucket side policies blocking. What's weird to me is that I have another function, not using DuckDB, using the same execution role, and it has no problems writing to S3. Any thoughts?

**Comment by alexanderluiscampino** [comment]
From my experience with libraries that abstract away calls to the AWS SDK, like in this case DuckDB -> S3, is that they might require more permissions than just a `PutObject`. You can try to enable the SDK full logging and see if you get more information on exactly which operation is being denied. Otherwise, give it full * permissions to see if it works, and then start taking them away one by one until you figure out which one it needs.
